compiler/rustc_hir/src/definitions.rs

```diff
@@ -245,6 +247,17 @@ impl DefPath {
 
         s
     }
+
+    pub fn get_impl_trait_in_trait_data(&self) -> Option<(DefId, Option<DefId>)> {
+        if let Some(def_path_data) = self.data.last()
+            && let DefPathData::ImplTraitInTrait(fn_def_id, of_trait) =
+            def_path_data.data
+        {
+            Some((fn_def_id, of_trait))
```

Comment:
Can this be implemented by walking `self.parent` until we find a `ValueNs`? As for `of_trait` I haven't seen it used anywhere, could it be computed from `fn_def_id`?